### ISSUE TITLE
FXP files which crash when mis-installed

### ISSUE THREAD

**Issue by tank-trax** [surge-synthesizer/surge#521]
I have identified a list of FXP files which crash the VST2 when loaded. I ran through the gamut and all other FXP files load without issue (there are a few that crackle a bit and few others that have no sound).

```
/surge/resources/data/patches_3rdparty/Nick Moritz/BASS - Chicago.fxp

/surge/resources/data/patches_factory/Bass/Bass three.fxp

/surge/resources/data/patches_3rdparty/inigo_kennedy_03/ik_dark_bass.fxp
/surge/resources/data/patches_3rdparty/inigo_kennedy_03/ik_distance_between_us.fxp
/surge/resources/data/patches_3rdparty/inigo_kennedy_03/ik_feint_memory.fxp
/surge/resources/data/patches_3rdparty/inigo_kennedy_03/ik_melodic_line_i.fxp
/surge/resources/data/patches_3rdparty/inigo_kennedy_03/ik_melodic_line_ii.fxp
/surge/resources/data/patches_3rdparty/inigo_kennedy_03/ik_melodic_line_iii.fxp
/surge/resources/data/patches_3rdparty/inigo_kennedy_03/ik_melodic_line_iv.fxp

/surge/resources/data/patches_factory/Sequence/Burial Ground.fxp

/surge/resources/data/patches_factory/Pad/Choirpad thing.fxp
/surge/resources/data/patches_factory/Pad/Computers in Space.fxp
/surge/resources/data/patches_factory/Pad/Hmm....fxp
/surge/resources/data/patches_factory/Pad/Pad 7.fxp
/surge/resources/data/patches_factory/Pad/Primes.fxp

/surge/resources/data/patches_factory/Polysynth/Embrrass.fxp
/surge/resources/data/patches_factory/Polysynth/Eyan.fxp
/surge/resources/data/patches_factory/Polysynth/Fonk.fxp
/surge/resources/data/patches_factory/Polysynth/Hombre.fxp
/surge/resources/data/patches_factory/Polysynth/Japanese Space-ulation Wheel.fxp
/surge/resources/data/patches_factory/Polysynth/Jim.fxp
/surge/resources/data/patches_factory/Polysynth/Past Tense.fxp
/surge/resources/data/patches_factory/Polysynth/Quantization Choice.fxp

/surge/resources/data/patches_factory/Pluck/Softsyncsaw.fxp
```

**Comment by esaruoho** [comment]
Thanks heaps @tank-trax for posting this list. This is some good stuff - let's see what shakes loose when we get to look at these :)

**Comment by baconpaul** [comment]
This is an outstandingly valuable contribution. Thank you very much

**Comment by baconpaul** [comment]
As discussed on slack, if assets are properly in ~/.local/Surge these do not crash

But they should not crash if assets are mis-installed so keep this open as a low priority item to investigate what the crash path is.

**Comment by baconpaul** [comment]
And linking this to #543

**Comment by baconpaul** [comment]
This one is the same. WaveTableOscillator does poorly with no wavetable. When loading choirpad stuff I see

```
#0  0x00007ffff1d9d88c in _mm_loadu_si128(long long __vector(2) const*) (__P=0x0) at /usr/lib/gcc/x86_64-linux-gnu/7/include/emmintrin.h:702
#1  WindowOscillator::ProcessSubOscs (this=0x3a48720, stereo=false) at src/common/dsp/WindowOscillator.cpp:187
#2  0x00007ffff1d9dcf2 in WindowOscillator::process_block (this=0x3a48720, pitch=42.1500015, drift=0, stereo=false, FM=false, fmdepth=0)
    at src/common/dsp/WindowOscillator.cpp:241
#3  0x00007ffff1ceb271 in COscillatorDisplay::draw (this=0x11cdaa0, dc=0x130c0e0) at src/common/gui/COscillatorDisplay.cpp:74
#4  0x00007ffff1cdc51a in VSTGUI::CView::drawRect (this=0x11cdaa0, pContext=0x130c0e0, updateRect=...) at vst3sdk/vstgui4/vstgui/lib/cview.h:65
#5  0x00007ffff1e0b448 in VSTGUI::CViewContainer::drawRect (this=0x1312a40, pContext=0x130c0e0, updateRect=...) at vstgui.surge/vstgui/lib/cviewcontainer.cpp:883
#6  0x00007ffff1dece84 in VSTGUI::CFrame::drawRect (this=0x1312a40, pContext=0x130c0e0, updateRect=...) at vstgui.surge/vstgui/lib/cframe.cpp:336
#7  0x00007ffff1df1065 in VSTGUI::CFrame::platformDrawRect (this=0x1312a40, context=0x130c0e0, rect=...) at vstgui.surge/vstgui/lib/cframe.cpp:1582
#8  0x00007ffff1eb51ab in VSTGUI::X11::Frame::Impl::redraw()::{lambda(VSTGUI::CDrawContext*, VSTGUI::CRect const&)#1}::operator()(VSTGUI::CDrawContext*, VSTGUI::CRect const&) const (__closure=0x7fffffffd6f8, context=0x130c0e0, rect=...) at vstgui.surge/vstgui/lib/platform/linux/x11frame.cpp:309
#9  0x00007ffff1eb6368 in VSTGUI::X11::DrawHandler::draw<std::vector<VSTGUI::CRect, std::allocator<VSTGUI::CRect> >, VSTGUI::X11::Frame::Impl::redraw()::{lambda(VSTGUI::CDrawContext*, VSTGUI::CRect const&)#1}>(std::vector<VSTGUI::CRect, std::allocator<VSTGUI::CRect> > const&, VSTGUI::X11::Frame::Impl::redraw()::{lambda(VSTGUI::CDrawContext*, VSTGUI::CRect const&)#1}) (this=0x1324268, dirtyRects=std::vector of length 155, capacity 256 = {...}, proc=...) at vstgui.surge/vstgui/lib/platform/linux/x11frame.cpp:152
#10 0x00007ffff1eb51d9 in VSTGUI::X11::Frame::Impl::redraw (this=0x1324240) at vstgui.surge/vstgui/lib/platform/linux/x11frame.cpp:308
#11 0x00007ffff1eb521e in VSTGUI::X11::Frame::Impl::invalidRect(VSTGUI::CRect)::{lambda()#1}::operator()() const (__closure=0x251fc68)
    at vstgui.surge/vstgui/lib/platform/linux/x11frame.cpp:323
```

if the wavetables are removed.

**Comment by baconpaul** [comment]
Same fix for #525 closes this one
